### PR TITLE
refactoring of topic test

### PR DESCRIPTION
**Pull request by gluk256**

### CODE REVIEW

**Comment by gluk256** [comment]
test execution time reduced

**Comment by gluk256** [comment]
never mind the errors -- they are not mine

File: test/libwhisper/whisperTopic.cpp [hunk 1]
```diff
@@ -314,39 +325,39 @@ BOOST_AUTO_TEST_CASE(topicAdvertising)
...
 	{
 		sessions = whost1->peerSessions();
-		if (!sessions.empty() && sessions.back().first->cap<WhisperPeer>()->bloom())
-			break;
-		else
-			this_thread::sleep_for(chrono::milliseconds(10));
+		if (!sessions.empty())
+		{
+			bf1 = sessions.back().first->cap<WhisperPeer>()->bloom();
+			if (bf1)
```
Comment (subtly) [inline test/libwhisper/whisperTopic.cpp]: Is this duplication of the !bf1 check in for()?

File: test/libwhisper/whisperTopic.cpp [hunk 2]
```diff
@@ -357,16 +368,18 @@ BOOST_AUTO_TEST_CASE(topicAdvertising)
...
 	{
 		sessions = whost2->peerSessions();
-		if (!sessions.empty() && sessions.back().first->cap<WhisperPeer>()->bloom())
-			break;
-		else
-			this_thread::sleep_for(chrono::milliseconds(10));
+		if (!sessions.empty())
+		{
+			bf2 = sessions.back().first->cap<WhisperPeer>()->bloom();
+			if (bf2)
```
Comment (subtly) [inline test/libwhisper/whisperTopic.cpp]: Move into for() just like previous test?

File: test/libwhisper/whisperTopic.cpp [hunk 3]
```diff
@@ -357,16 +368,18 @@ BOOST_AUTO_TEST_CASE(topicAdvertising)
...
+				break;
+		}
+
+		this_thread::sleep_for(chrono::milliseconds(10));
```
Comment (subtly) [inline test/libwhisper/whisperTopic.cpp]: step instead of 10?

**Pull request merged**